src/static/login.html

```diff
@@ -325,6 +391,20 @@
                 xhr.send();
             }
 
+            function login(user, password) {
+                var headers = {
+                    "Authorization": "Basic " + btoa(utf8(user + ":" + password))
+                };
+                send_login_request("GET", headers, false);
+            }
+
+            function converse(id, msg) {
+                var headers = {
+                    "Authorization": "Login-Reply " + id + " " + btoa(utf8(msg))
+                };
+                send_login_request("GET", headers, true);
```

Comment:
Yep, I felt weird about this too. Originally i did it with POST, but cockpit goes out of it's way to only allow GET requests with empty bodies. I was hesitant to remove that restriction . But if you think that's better I still have the code lying around somewhere.